torch/_inductor/ir.py

```diff
@@ -3250,12 +3250,6 @@ def set_cpp_kernel(self, kernel):
         assert (
             not kernel._schema.is_mutable
         ), f"mutable {kernel.__name__} is not supported with cpp_wrapper"
-        assert all(
-            x.alias_info is None for x in kernel._schema.arguments
-        ), f"{kernel.__name__} with alias_info arguments is not supported with cpp_wrapper"
-        assert all(
-            x.alias_info is None for x in kernel._schema.returns
-        ), f"{kernel.__name__} with alias_info returns is not supported with cpp_wrapper"
```

Comment:
We didn't support alias_info for now. Take an OP of arguments and returns with alias_info as an example:
```yaml
linalg_eig.out(Tensor self, *, Tensor(a!) eigenvalues, Tensor(b!) eigenvectors) -> (Tensor(a!) eigenvalues, Tensor(b!) eigenvectors)
```

`Tensor(a!) eigenvalues` and `Tensor(b!) eigenvectors` have alias info. In the cpp function of this OP, the arg type will be `Tensor&` instead of `const Tensor&` for the input that has alias info. The output will be `Tensor&` instead of `Tensor`.
```cpp
std::tuple<Tensor&, Tensor&> linalg_eig_out(const Tensor& input, Tensor& values, Tensor& vectors)
```

This logic is not enabled yet thus I've added this check here.

Reply:
@davidberard98 which op requires you to remove the checking here?

Reply:
First, thanks for the review!

In this case, I don't think we actually need `Tensor&` in order to deal with aliasing: I tested it with this:

```c++
  at::Tensor x = at::tensor({1.0, 2.0, 3.0, 4.0});
  at::Tensor y = at::tensor({1.0, 2.0, 3.0, 4.0});
  at::Tensor out = at::tensor({0.0});
  at::Tensor& ret = at::add_out(x, y, out);
  ASSERT_TRUE((&out) == (&ret)); // this fails
```

My interpretation from this is that the tensor returned from add_out is a different tensor from the out tensor, even though they are both aliasing the same tensor storage. Therefore, I think it's safe to assign by value instead of by reference when accepting the return value from an op that has aliasing info.

Please let me know if I'm missing something here!

Reply:
@desertfire it's for view_as_real

Reply:
How about updating the check to be:
We don't support the below case in cpp_wrapper:
`x.alias_info is not None and x.alias_info.is_write`

Reply:
@chunyuan-w sure, that makes sense

Reply:
According to the comment https://github.com/pytorch/pytorch/pull/103183/files#r1222387067, `Tensor` is fine for both view and mutation, do we still need `is_write` check?

Reply:
@jgong5, I don't know if we need the `is_write` check - I was just going off of @chunyuan-w 's comment. However,  I think that functionalization during AOT Autograd might remove any ops that contain mutation, so it might not matter.

For now I'll add a comment here, unless you or @chunyuan-w feel that it would be better to completely remove the check

Reply:
@chunyuan-w oh, you're right - I'll update the comment.